core/src/main/java/org/elasticsearch/action/admin/indices/create/CreateIndexRequest.java

```diff
@@ -440,6 +443,28 @@ public CreateIndexRequest updateAllTypes(boolean updateAllTypes) {
         return this;
     }
 
+    public int waitOnActiveShards() {
+        return waitOnActiveShards;
+    }
+
+    /**
+     * Sets the number of shard copies that must be active for the write to proceed.
+     * Defaults to 1, which means only the primary is required to be active.  Set this
+     * value to -1 to ensure all shards (primary and all replicas) are active before
+     * proceeding with the write.  A value of 0 implies the value is not set, so the
+     * default will be used.
+     *
+     * @param waitOnActiveShards number of active shard copies to wait on, must be between 0 and numShardCopies, or -1 for all.
+     */
+    public CreateIndexRequest waitOnActiveShards(int waitOnActiveShards) {
+        if (waitOnActiveShards == 0 || waitOnActiveShards < -1) {
```

Comment:
Why is it not possible to specify 0? I might want to create an index without waiting for any shard of that index to be active.